scrapy/core/downloader/tls.py

```diff
@@ -48,6 +48,11 @@ def _identityVerifyingInfoCallback(self, connection, where, ret):
                         'Remote certificate is not valid for hostname "{}"; {}'.format(
                             self._hostnameASCII, e))
 
+                except ValueError as e:
+                    logger.warning(
+                        'SSL/TLS verification failed for hostname "{}"; {}'.format(
+                            self._hostnameASCII, e))
```

Comment:
OK.
What about "host" or "netloc"?
And something like "Ignoring error while verifying certificate from host {}
(exception: {})" maybe,
with target host and exception
Le 13 juil. 2016 12:28 AM, "Mikhail Korobov" <notifications@github.com> a
écrit :

> In scrapy/core/downloader/tls.py
> <https://github.com/scrapy/scrapy/pull/2094#discussion_r70534245>:
>
> > @@ -48,6 +48,11 @@ def _identityVerifyingInfoCallback(self, connection, where, ret):
> >                          'Remote certificate is not valid for hostname "{}"; {}'.format(
> >                              self._hostnameASCII, e))
> >
> > +                except ValueError as e:
> > +                    logger.warning(
> > +                        'SSL/TLS verification failed for hostname "{}"; {}'.format(
> > +                            self._hostnameASCII, e))
>
> @redapple <https://github.com/redapple> I have 2 reservations about the
> error message:
>
>    1. 'certificate verification failure' can be read as 'wrong
>    certificate' or a 'misconfigured remote website' or 'wrong configuration of
>    local verification software'. But here it is not a bad certificate or old
>    openssl or a problem with a remote website, it is just a bug in software;
>    because of a bug the verification is not even attempted. I don't know how
>    to express that well in error message though.
>    2. it says 'hostname {}'.format(self._hostnameASCII), but
>    _hostnameASCII is not a hostname here - it is an IP, and this is precisely
>    what causes the error.
>
> I think we shouldn't describe the actual bug in the error message because
> ValueError is too broad - it could happen other issue will cause ValueError
> in future, or maybe it can be triggered now by something else.
>
> What about "Ignoring certificate verification error for {};
> {}".format(self._hostnameASCII, e)? It doesn't fully address (1).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scrapy/scrapy/pull/2094/files/37efdde3e3dd4efecf7dccd361fceab6c851d0ff#r70534245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA2GGMc4yG_0sC7jmUU2LrN5jkjsGJLGks5qVBUMgaJpZM4JGBi->
> .
>